__tests__/middleware.js

```diff
@@ -0,0 +1,51 @@
+import {createInstance} from 'osjs';
+import Middleware from '../src/middleware.js';
+
+describe('Middleware', () => {
+  let core;
+
+  beforeAll(() => {
+    return createInstance()
+      .then(c => (core = c));
+  });
+
+  afterAll(() => core.destroy());
+
+  test('#add', () => {
+    const middleware = new Middleware();
+    const callback = () => {};
+
+    middleware.add('group', callback);
+
+    expect(middleware.get('group')[0]).toEqual(callback);
+  });
+
+  test('#remove', () => {
+    const middleware = new Middleware();
+    const callback = () => {};
+
+    middleware.add('group', callback);
+    middleware.remove('group', callback);
+
+    expect(middleware.get('group').length).toEqual(0);
+  });
+
+  test('#clear', () => {
+    const middleware = new Middleware();
+    const callback = () => {};
+
+    middleware.add('group', callback);
+    middleware.clear();
+
+    expect(middleware.get('group').length).toEqual(0);
+  });
+
+  test('#core middleware', () => {
+    const callback = () => {};
+    core.middleware('osjs/service', callback);
+
+    const middleware = core.make('osjs/middleware').get('osjs/service');
+
+    expect(middleware[0]).toEqual(callback);
+  });
+});
```

Comment:
Maybe add a test for `get` as well here ?

Reply:
It would be like the test for `add`, isn't it?

Reply:
I was thinking of having a test that loops over the callback list and checks if they all were called correctly, ex:

```
const callback = jest.fn()
// ...
expect(callback.mock.calls.length).toBe(1)
```